app/src/main/java/com/example/task1/ui/dashboard/DashboardViewModel.kt

```diff
@@ -0,0 +1,54 @@
+package com.example.task1.ui.dashboard
+
+import ApiService
+import android.content.Context
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.liveData
+import com.example.task1.data.models.Banner
+import com.example.task1.data.models.Items
+import com.example.task1.data.models.Results
+import com.example.task1.data.repository.RepositoryImpl
+import com.example.task1.data.repostory.Repository
+import kotlinx.coroutines.Dispatchers
+
+
+
+
+
+
+class DashboardViewModel(context: Context) : ViewModel() {
+
+    private val repository: Repository
+
+    init {
+        // Define headers here
+        val headers = mapOf(
+            "Authorization" to "eyJhbGciOiJIUzM4NCJ9.eyJzdWIiOiJhcHBsaWNhdGlvbiIsInVpZCI6IkNzcDZUTDk0Ui02cHFmbnZvQVk4RXciLCJ1c2VySWQiOiI2NTU3MGZiMDljZWYyYjZlOTVhNzRiNzgiLCJyb2xlcyI6WyJVU0VSIl0sImFwaVR5cGUiOiJBUFAiLCJ0YWdzIjpbIk1BUktFVF9BUFBfVVNFUiJdLCJpYXQiOjE3MDUzMjQ4NjksInRlbmFudCI6IlhMbjJXSXhkV3gwREJ0IiwidmVyc2lvbiI6Nn0.iUG0lSxkm1I6uU9uzBzKDHE9kJ63_vVYjaXQJvqylghwBd_SSBPhMNiHEGL27_2Y",
+            "Content-Type" to "application/json"
+        )
+
+        val apiService = RetrofitClient.getClient(headers).create(ApiService::class.java)
```

Comment:
Why is retrofitClient being initialised in ViewModel? Its needed by repository so it should be initialised there not in ViewModel.

Reply:
Yes it is also done.